apps/condo/domains/notification/adapters/redStoreAdapter.js

```diff
@@ -0,0 +1,265 @@
+const { faker } = require('@faker-js/faker')
+const { isEmpty, isNull, get, isObject } = require('lodash')
+
+const conf = require('@open-condo/config')
+const { getLogger } = require('@open-condo/keystone/logging')
+
+const { RedStoreNotificationSender } = require('@condo/domains/notification/adapters/redStore/redStoreNotificationSender')
+const {
+    PUSH_FAKE_TOKEN_SUCCESS,
+    PUSH_FAKE_TOKEN_FAIL,
+    REDSTORE_CONFIG_ENV,
+    PUSH_TYPE_DEFAULT,
+    PUSH_TYPE_SILENT_DATA,
+    APPS_WITH_DISABLED_NOTIFICATIONS_ENV,
+} = require('@condo/domains/notification/constants/constants')
+const { EMPTY_REDSTORE_CONFIG_ERROR, EMPTY_NOTIFICATION_TITLE_BODY_ERROR } = require('@condo/domains/notification/constants/errors')
+
+const REDSTORE_CONFIG = conf[REDSTORE_CONFIG_ENV] ? JSON.parse(conf[REDSTORE_CONFIG_ENV]) : null
+const APPS_WITH_DISABLED_NOTIFICATIONS = conf[APPS_WITH_DISABLED_NOTIFICATIONS_ENV] ? JSON.parse(conf[APPS_WITH_DISABLED_NOTIFICATIONS_ENV]) : []
+const DEFAULT_PUSH_SETTINGS = {
+    apns: { payload: { aps: { 'mutable-content': 1, sound: 'default' } } },
+}
+const HIGH_PRIORITY_SETTINGS = { android: { priority: 'high' } }
+
+const logger = getLogger('firebaseAdapter')
+
+/**
+ * Send push notification to pushToken via app, configured by RedStore_CONFIG in .helm (.env)
+ * Attention! Notifications could only be sent to devices, connected via same PROJECT_ID.
+ * Attempts to send push notifications to devices, connected through different projects will fail.
+ */
+class RedStoreAdapter {
+    projectId = null
+    #config = null
+
+    constructor (config = REDSTORE_CONFIG) {
+        try {
+            if (isEmpty(config)) throw new Error(EMPTY_REDSTORE_CONFIG_ERROR)
+        } catch (error) {
+            // For CI/local tests config is useless because of emulation via FAKE tokens
+            logger.error({ msg: 'redStore adapter error', error })
+        }
+
+        this.projectId = get(config, 'project_id', null)
+        this.#config = config
+        // not user input. No ReDoS regexp expected
+        // nosemreg: javascript.lang.security.audit.detect-non-literal-regexp.detect-non-literal-regexp
+        this.messageIdPrefixRegexp = new RegExp(`projects/${this.projectId}/messages`)
+    }
+
+    /**
+     * Firebase rejects push if any of data fields is not a string, so we should convert all non-string fields to strings
+     * @param data
+     */
+    static prepareData (data = {}, token) {
+        const result = { token }
+
+        for (const key in data) {
+            if (data.hasOwnProperty(key)) result[key] = String(data[key])
+        }
+
+        return result
+    }
+
+    /**
+     * Validates and prepares notification significant fields
+     * @param title
+     * @param body
+     * @returns {{title, body}}
+     */
+    static validateAndPrepareNotification ({ title, body } = {}) {
+        if (!title || !body || isEmpty(title) || isEmpty(body)) throw new Error(EMPTY_NOTIFICATION_TITLE_BODY_ERROR)
```

Comment:
It's preferable to hardcode the error message directly in the code. Moving it to a separate file can make it more challenging to review and understand the error context, ultimately reducing code readability. Keeping the error message inline enhances clarity and makes it easier for others to follow the logic.